transformer_engine/pytorch/float8_tensor.py

```diff
@@ -381,7 +382,8 @@ def __new__(
             raise ValueError(
                 "Float8Tensor requires non-differentiable data buffer"
             )
-        data = data.cuda()
+        if not data.is_cuda:
+            data = data.cuda()
```

Comment:
Is this not a noop if tensor is already on device?

Reply:
Yep, tensor conversion functions (`to`, `cuda`, `contiguous`, `float`) will be no-ops if possible. I prefer being as paranoid as possible until CPU overheads start becoming a problem.

Reply:
Ah, I missed the "not" in your question. The conversion functions add some CPU overhead, so this is a microoptimization.